Extensions/Today/TodayViewController.swift

```diff
@@ -9,123 +9,169 @@ import SnapKit
+        let imageButton = ImageButtonWithLabel()
+        imageButton.addTarget(self, action: #selector(onPressNewPrivateTab), forControlEvents: .TouchUpInside)
+
+        let button = imageButton.button
         button.setImage(UIImage(named: "new_private_tab_button_normal"), forState: .Normal)
         button.setImage(UIImage(named: "new_private_tab_button_highlight"), forState: .Highlighted)
-        return button
+
+        let label = imageButton.label
+        label.text = NSLocalizedString("TodayWidget.NewPrivateTabButtonLabel", value: "New Private Tab", tableName: "Today", comment: "New Private Tab button label")
+        label.textColor = TodayUX.privateBrowsingColor
+        label.font = UIFont.systemFontOfSize(TodayUX.imageButtonTextSize)
+
+        return imageButton
     }()
 
-    private lazy var openURLFromClipboardView: UIView = {
-        let view = UIView()
-        let button = UIButton()
+    private lazy var openCopiedLinkButton: ButtonWithSublabel = {
+        let button = ButtonWithSublabel()
+        
         button.setTitle(NSLocalizedString("TodayWidget.GoToCopiedLinkLabel", value: "Go to copied link", tableName: "Today", comment: "Go to link on clipboard"), forState: .Normal)
         button.addTarget(self, action: #selector(onPressOpenClibpoard), forControlEvents: .TouchUpInside)
+        button.setBackgroundColor(TodayUX.backgroundHightlightColor, forState: .Highlighted)
+
+        button.label.font = UIFont.systemFontOfSize(TodayUX.labelTextSize)
+        button.setImage(UIImage(named: "copy_link_icon"), forState: .Normal)
+
+        button.subtitleLabel.font = UIFont.systemFontOfSize(TodayUX.linkTextSize)
 
-        return view
+        return button
     }()
 
+    private lazy var buttonContainer: UIView = UIView()
+
+    private var copiedURL: NSURL? {
+        if let string = UIPasteboard.generalPasteboard().string,
+            url = NSURL(string: string) where url.isWebPage() {
+            return url
+        } else {
+            return nil
+        }
+    }
+
+    private var hasCopiedURL: Bool {
+        return copiedURL != nil
+    }
+
     override func viewDidLoad() {
         super.viewDidLoad()
 
-        let buttonContainer = UIView()
         view.addSubview(buttonContainer)
 
-//        view.backgroundColor = UIColor.blueColor()
-//        buttonContainer.backgroundColor = UIColor.redColor()
-
-        buttonContainer.snp_makeConstraints { make in
-            make.width.equalTo(view.snp_width).multipliedBy(0.8)
-            make.centerX.equalTo(view.snp_centerX)
-        }
-
         // New tab button and label.
         buttonContainer.addSubview(newTabButton)
         newTabButton.snp_makeConstraints { make in
-            make.topMargin.equalTo(10)
-            make.leading.equalTo(buttonContainer.snp_leading).offset(44)
-            make.height.width.equalTo(44)
+            make.top.equalTo(buttonContainer.snp_top)
+            make.left.equalTo(buttonContainer.snp_left)
```

Comment:
if you're making things equal to the same value as another thing, you don't have to be quite so explicit, so
`make.top.left.equalTo(buttonContainer)` will have the same effect as both of these lines